File: LeetcodeQuestions/leetcode_challenges/February_challange/simplify_path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        chars = path.split("/")
        stack = []
        con_path = "/"
        special = set([".","..",""])
        for char in chars:
            if len(stack) and char == "..":
                stack.pop()
            elif char not in special:
                stack.append(char)
        con_path += "/".join(stack)
        return con_path

File: LeetcodeQuestions/leetcode_challenges/February_challange/simplifyPath.js
function simplifyPath(path) {
    let stack = [];
    let chars = path.split("/");
    let conPath = "/";
    let special = new Set([".", "..", ""]);

    for (let char of chars) {
        if (stack.length && char === "..") {
            stack.pop();
        } else if (!special.has(char)) {
            stack.push(char);
        }
    }
    return conPath += stack.join("/");
}

console.log(simplifyPath("/home/"))
console.log(simplifyPath("/../"))
console.log(simplifyPath("/home//foo/"))
console.log(simplifyPath("/a/./b/../../c/"))